[fx] Do not add Proxy on Tensor

yushangdi opened the pull request:
Summary: Switch to set_proxy_slot instead of set the proxy directly on the Tensor. We do not want to add Proxy to tensor objects, because Proxy cannot be deepcopied or pickeled and can cause problems when users want to deepcopy or pickle models.

Test Plan: CI

Differential Revision: D61277650

zou3519 reviewed (approved):
Don't forget to change the commit message

yushangdi commented:
> Don't forget to change the commit message

it's updated on the internal Diff. Not sure why the PR is not updated. I just tried to export again.

yushangdi committed:
[fx] Do not add Proxy on Tensor (#133470)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/133470

We do not want to add Proxy to tensor objects, because Proxy cannot be deepcopied or pickeled and can cause problems when users want to deepcopy or pickle models.

The DCE pass on fx graph now looks at op schema to avoid removing in-place ops, so we do not need to rely on the added Proxy anyway.

Test Plan: CI

Differential Revision: D61277650
